core/src/main/scala/spinal/core/internals/ComponentEmitterVerilog.scala

```diff
@@ -1732,4 +1740,21 @@ end
   fillExpressionToWrap()
   emitEntity()
   emitArchitecture()
+
+  def signalNoUse(sig: BaseType): Boolean = {
+    var noUse = true
+    component.dslBody.walkStatements {
```

Comment:
Ahh i would say this doesn't scale well. for each request it will check the whole component dslBody. 
I would say, the best would be to generate a noUse linkedhashmap that specify for every output of the current component if it is used once. And then use that linkedhashmap for